src/core/core.datasetController.js

```diff
@@ -287,6 +374,99 @@ helpers.extend(DatasetController.prototype, {
 		});
 	},
 
+	/**
+	 * @private
+	 */
+	_storeParsedData: storeParsedData,
+
+	/**
+	 * @private
+	 */
+	_parse: function(start, count) {
+		var me = this;
+		var meta = me.getMeta();
+		var data = me._data;
+		var i, ilen, item;
+
+		if (!data || data.length === 0) {
+			return;
+		}
+
+		for (i = start, ilen = start + count; i < ilen; ++i) {
+			item = data[i];
+			if (helpers.isArray(item)) {
+				me._parseArrayData(item, meta, i);
+			} else if (helpers.isObject(item)) {
+				me._parseObjectData(item, meta, i);
+			} else {
+				me._parsePlainData(item, meta, i);
+			}
+		}
+	},
+
+	/**
+	 * @private
+	 */
+	_parsePlainData: function(value, meta, index) {
+		var iScale = this._getIndexScale();
+		var vScale = this._getValueScale();
+		var labels = iScale._getLabels() || [];
+		var v = vScale._parse(value, index);
+		var iv = iScale !== vScale && index < labels.length
+			? iScale._parse(labels[index], index)
+			: index;
+		storeParsedData(meta, index, iScale, vScale, iv, v);
+	},
+
+	/**
+	 * @private
+	 */
+	_parseArrayData: function(arr, meta, index) {
+		var xScale = this.getScaleForId(meta.xAxisID);
```

Comment:
Why do `_parseArrayData` and `_parseObjectData` use `xScale` and `yScale` while `_parsePlainData` uses `iScale` and `vScale`? Can they be made consistent?

Reply:
They can't, because for example:
```js
type: 'horizontalBar',
data: {
  datasets: [{
    data: [{x: 1, y: 2}, {x: 2, y: 5}]
  }
},
options: {
  scales: {
    xAxes: [{id: 'x', type: 'linear'}],
    yAxes: [{id: 'y', type: 'linear'}]
  }
}
```

_getIndexScaleId() would return 'x' and _getValueScaleId() would return 'x'.
if type was changed to 'bar', value/index scale would swap. The parsing needs to be done same way in both cases though.

Reply:
Do you really mean that they both would return `'x'` or did you mean that one would return `'y'`?

Reply:
Fixed the comment